week-3/Homework/G-extra/2-map.js

```diff
@@ -11,10 +11,18 @@ var attendanceCounts = [
   ["Elamin", 6],
   ["Adam", 7],
   ["Tayoa", 11],
-  ["Nina", 10]
+  ["Nina", 10],
 ];
 
-var eligibleStudentNames; // TODO: Complete this line.
+var eligibleStudentNames = attendanceCounts
+  .map((pair) => {
+    let name = pair[0];
+    let attendance = pair[1];
+    if (attendance >= 8) return name;
+  })
+  .filter((name) => {
+    return name != undefined;
+  }); // TODO: Complete this line.
 
```

Comment:
This is similar to the previous example, that its better to apply filter first, then map the filtered array to the values that you want.

Reply:
You are absolutely right. My first code looks not optimal, I have changed it. Thanks for your feedback.